airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/response_models.py

```diff
@@ -22,10 +25,34 @@ class ErrorResolution:
     error_message: Optional[str] = None
 
 
-DEFAULT_ERROR_RESOLUTION = ErrorResolution(
-    response_action=ResponseAction.RETRY,
-    failure_type=FailureType.system_error,
-    error_message="The request failed due to an unknown error.",
-)
+def _format_exception_error_message(exception: Exception) -> str:
+    return f"{type(exception).__name__}: {str(exception)}"
+
+
+def _format_response_error_message(response: requests.Response) -> str:
+    try:
+        response.raise_for_status()
+    except HTTPError as exception:
+        return filter_secrets(f"Response was not ok: `{str(exception)}`. Response content is: {response.text}")
+    # We purposefully do not add the response.content because the response is "ok" so there might be sensitive information in the payload.
+    # Feel free the
+    return f"Unexpected response with HTTP status {response.status_code}"
+
+
+def create_fallback_error_resolution(response_or_exception: Optional[Union[requests.Response, Exception]]) -> ErrorResolution:
+    if response_or_exception is None:
+        # We do not expect this case to happen but if it does, it would be good to understand the cause and improve the error message
+        error_message = "Unexpected argument while handling error"
```

Comment:
nit: let's be more explicit here since we're pretty confident this case is a mistake and say `"Error handler did not receive a valid response or exception. This is unexpected please contact Airbyte Support". 

Or something akin to that. I think it would just be helpful on our side to know that this was rooted in the error handler since the original message is a bit vague

Reply:
I like the suggestion. Thanks!